Firmware/tmc2130.cpp

```diff
@@ -883,14 +885,115 @@ void tmc2130_get_wave(uint8_t axis, uint8_t* data)
 	tmc2130_set_pwr(axis, pwr);
 }
 
+// Calculate constant torque value for a given microstep positionAdd commentMore actions
+//
+// Maintains |A|² + |B|² = constant throughout the microstep cycle, where A and B are the two motor phases. 
+// Uses a two-phase approach: positions 0-127 follow a power-corrected sine curve, 
+// while positions 128-255 are calculated using the constant torque constraint equation.
+//
+// Creates only deltas achievable by TMC2130 in at most four segments. It relies on the fact, 
+// that for power corrected sine curve with fac [1, 1.2] the slope stays in the range [0, 2]
+// and fits at most three segments.
+//
+// Parameters:
+// - i: microstep position (0-255)
+// - va: previous amplitude value for delta calculation
+// - fac: power factor for linearity correction
+// - tcorr: pre-calculated torque correction factor
+// - carry: quantization error carry-forward (modified by reference)
+// - prev_theoretical_value: previous theoretical value for slope calculation (modified by reference)
+//
+// Returns: 8-bit amplitude value clamped to [SIN0, AMP] range
+//
+// References:
+// - Prusa Forum: "TMC2130 constant torque algorithm" discussion
+//   https://forum.prusa3d.com/forum/original-prusa-i3-mk3s-mk3-user-mods-octoprint-enclosures-nozzles/stepper-motor-upgrades-to-eliminate-vfa-s-vertical-fine-artifacts/paged/2/
+// - Analog Devices AN-026: "Stepper Motor Control Using TMC2130"
+//   https://www.analog.com/en/resources/app-notes/an-026.html
+uint8_t tmc2130_calc_constant_torque_value(uint8_t i, uint8_t va, float fac, float tcorr,
+                                          float& prev_theoretical_value) {
+	constexpr uint8_t SIN0 = 0;
+	constexpr uint8_t AMP = 248;  // Amplitude limit as per AN-026 recommendation
+	constexpr float TARGET_MAGNITUDE_SQUARED = (float)AMP * AMP + (float)SIN0 * SIN0;
+
+	// Theoretical constant torque value at microstep position i
+	float theoretical_value;
+
+	if (i < 128) {
+		// Phase 1 (positions 0-127): Power-corrected sine curve
+		// Calculate theoretical value using sine function with power factor
+		// correction and tcorr adjustment for midpoint matching
+		float sin_val = sin(M_PI * (float)i / 512.0f);
```

Comment:
I suspect the compiler doesnt calculate `M_PI / 512.f` constant at build time.

If I'm right this will consume less memory and drop one division operation:
```cpp
constexpr float sin_gain = M_PI / 512.f;
float sin_val = sin(sin_gain * i);
```

I may be wrong but I think I have seen this happen before.